File: src/collatz.rs
fn collatz(x: int) -> int {
    println!("{}", x);
    if x == 1 { 1 }
    else if x % 2 == 1 { collatz(3 * x + 1) }
    else { collatz(x / 2) }
}

fn main() {
    println!("{}", collatz(9));
}

File: src/collatz.cpp
#include <iostream>

int collatz(int n)
{
        std::cout << n << std::endl;
        if (n <= 1)
                return 1;
        else
        {
                if (n % 2 == 0)
                        return collatz(n / 2);
                else
                        return collatz(n * 3 + 1);
        }
}

int main()
{
        collatz(9);
}
